src/main/java/org/elasticsearch/index/query/QueryBuilder.java

```diff
@@ -45,6 +45,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     }
 
     /**
+     * @return a unique name this query is identified with
+     */
+    abstract public String queryId();
```

Comment:
yea thanks for the link, correct order is public abstract, which I think we've been using in the rest of the codebase.